src/shared/initialize.ts

```diff
@@ -0,0 +1,9 @@
+import { GetSiteResponse } from 'lemmy-js-client';
+import { UserService } from './services';
+import { i18n } from './i18next';
+import { getLanguage } from './utils';
+
+export const initializeSite = (site: GetSiteResponse) => {
```

Comment:
I'm not a fan of closures where they aren't necessary. Instead of this new file, just move this function to `utils.ts`, and 

`export function initializeSite(site: GetSiteResponse)...`

Reply:
I'll change to a named function. Babel does give the closure a name in this case though, just FYI.

Regarding utils.ts, I think it's very busy. I'd much rather see a `utils` directory with smaller, more categorized files. I can make a detailed suggestion of what I have in mind in an issue for discussion if you want

Reply:
Sure, I agree utils is a bit busy. A lot of things in there are only used by a single component, in which case they should probably just be private functions there. 

But overall its nice to only have to worry about a single import for logic functions that aren't components.

Reply:
Switched to function. The main problem right now is that if I move initializeSite into utils, there are dependency cycles which breaks getLanguage(), due to initializeSite needing to import `i18next`. That's another drawback of keeping all the utils in the same file.